### DISCUSSION TITLE
Externalise the builds?

### DISCUSSION THREAD

**Discussion by prologic** [goreleaser/goreleaser#1835] [Q&A] [answered]
Hello Gophers!

I have a project over at https://github.com/jointwt/twtxt and since we made a decision to embed the use of a C library that uses Swig bindings for internal ffmpeg and webm support, we are unable to use the GoReleaser tool and config we have setup to push our new releases.

I haven't yet figured out a way to push forward, but my question is quite simple; Is there a way I can externalise the builds rather than relying on the cross-compilation tooling included in `go build` which `goreleaser` also uses?

The main problems I've run into are to do with the fact it is hard to cross-compile Swig/C bindings from my build machine -- mac OS.

Thanks!

**Comment by prologic** [comment]
For some context on this I filed [this issue](https://github.com/chai2010/webp/issues/33) that has examples of the problems I've run in to...

**Comment by caarlos0** [comment, marked as answer]
We have a guide for cross-compiling with CGO on our website, although its provided by a third party: https://goreleaser.com/cookbooks/cgo-and-crosscompiling/

**Reply by prologic** [reply]
That's the thing. I want to avoid cross-compiling altogether. Its actually problematic for me. Does this guide help with that?

**Reply by prologic** [reply]
Ideally I'd create the release with `goreleaser` then let Github Actions workflows kick off builds (after the release is compelted) and build on Ubuntu, macOS natively without cross-compliing, and theen  upload the binary artifacts to the release.

**Reply by caarlos0** [reply]
GoReleaser does not have a native way of doing that.

I think you'll need to create one goreleaser config for each platform and run goreleaser multiple times (one for each platform), but I don't have example configuration for that and the guide I pointed you to won't help you with that either... 

If you manage to make it work, it would be a nice contribution to our cookbooks section for sure 🙏
